src/components/Main/Main.tsx

```diff
@@ -41,19 +41,28 @@ async function runSimulation(
   severity: Severity[],
   setResult: React.Dispatch<React.SetStateAction<AlgorithmResult | undefined>>,
   setEmpiricalCases: React.Dispatch<React.SetStateAction<EmpiricalData | undefined>>,
+  setIsRunning: React.Dispatch<React.SetStateAction<boolean>>,
 ) {
-  const paramsFlat = {
-    ...params.population,
-    ...params.epidemiological,
-    ...params.simulation,
-    ...params.containment,
-  }
+  setIsRunning(true)
+  try {
+    const paramsFlat = {
+      ...params.population,
+      ...params.epidemiological,
+      ...params.simulation,
+      ...params.containment,
+    }
 
-  const caseCounts = getCaseCountsData(params.population.cases)
-  const newResult = await run(paramsFlat, severity, scenarioState.ageDistribution)
-  setResult(newResult)
-  caseCounts.sort((a, b) => (a.time > b.time ? 1 : -1))
-  setEmpiricalCases(caseCounts)
+    const caseCounts = getCaseCountsData(params.population.cases)
+    const newResult = await run(paramsFlat, severity, scenarioState.ageDistribution)
+    setResult(newResult)
+    caseCounts.sort((a, b) => (a.time > b.time ? 1 : -1))
+    setEmpiricalCases(caseCounts)
+  } catch (error) {
+    // Rejected promises are thrown by await functions.
+    // Catch and log the error message to console.
+    console.error(error)
+  }
+  setIsRunning(false)
```

Comment:
This code is essentially the same as before, except that the `run` call is being transparently executed on a webworker. The try/catch block is necessary to catch any errors thrown by the `await`.  The `run` function isn't expected to throw errors, but the webworker might – such as warnings about structured clone failures or exceptions in the worker code.